source/interaction/interaction_utilities.cc

```diff
@@ -1132,6 +1133,387 @@ namespace fdl
       }
 #undef ARGUMENTS
   }
+  bool
+  intersect_line_with_edge(std::vector<std::pair<double, Point<1>> >& t_vals,
+                           const DoFHandler<1, 2>::active_cell_iterator elem,
+                           const dealii::MappingQ<1, 2> &mapping,
+                           dealii::Point<2> r,
+                           dealii::Tensor<1,2> q,
+                           const double tol)
+  {
+    bool is_interior_intersection = false;
+    t_vals.resize(0);
+    elem->get_triangulation().get_manifold(1);
+    int d_intersection_refinement =5;
+    switch (mapping.get_degree())
+      {
+        case 1:
+          {
+            // Linear interpolation:
+            //
+            //    0.5*(1-u)*p0 + 0.5*(1+u)*p1 = r + t*q
+            //
+            // Factor the interpolation formula:
+            //
+            //    0.5*(p1-p0)*u+0.5*(p1+p0) = r + t*q
+            //
+            // Solve for u:
+            //
+            //    a*u + b = 0
+            //
+            // with:
+            //
+            //    a = 0.5*(-p0+p1)
+            //    b = 0.5*(p0+p1) - r
+            const dealii::Point p0 = elem->vertex(0);
+            const dealii::Point p1 = elem->vertex(1);
```

Comment:
Even though this works with C++17 we should still use `Point<spacedim>` here and below.